sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -179,6 +181,16 @@ def _fit(
         if not fh.is_all_out_of_sample(cutoff=self.cutoff):
             raise NotImplementedError("in-sample prediction is currently not supported")
 
+        if self.freq == "auto":
```

Comment:
This one is a biased suggestion based on just ny personal preferences.

I suggest you first check for failure case, i.e. `if self.freq == "auto" and fh.freq is None` and raise if that is the case. Else proceed as usual.

Reply:
do we always need a `freq` in `neuralforecast`, or can it be `None` upstream?

Reply:
from `neuralforecast.NeuralForecast`
```py
class NeuralForecast:
    def __init__(
        self, models: List[Any], freq: str, local_scaler_type: Optional[str] = None
    ):
```
I guess we always need `freq`, it can't be `None`